test/core/slice/slice_test.cc

```diff
@@ -221,73 +227,217 @@ static void test_slice_split_tail_works(size_t length) {
+                         }()),
+                         [](const testing::TestParamInfo<size_t>& info) {
+                           return std::to_string(info.param);
+                         });
+
+TEST(GrpcSliceTest, SliceFromCopiedString) {
   static const char* text = "HELLO WORLD!";
   grpc_slice slice;
 
-  LOG_TEST_NAME("test_slice_from_copied_string_works");
-
   slice = grpc_slice_from_copied_string(text);
-  GPR_ASSERT(strlen(text) == GRPC_SLICE_LENGTH(slice));
-  GPR_ASSERT(
-      0 == memcmp(text, GRPC_SLICE_START_PTR(slice), GRPC_SLICE_LENGTH(slice)));
+  EXPECT_EQ(strlen(text), GRPC_SLICE_LENGTH(slice));
+  EXPECT_EQ(
+      0, memcmp(text, GRPC_SLICE_START_PTR(slice), GRPC_SLICE_LENGTH(slice)));
   grpc_slice_unref_internal(slice);
 }
 
-static void test_moved_string_slice(void) {
-  LOG_TEST_NAME("test_moved_string_slice");
-
+TEST(GrpcSliceTest, MovedStringSlice) {
   // Small string should be inlined.
   constexpr char kSmallStr[] = "hello12345";
   char* small_ptr = strdup(kSmallStr);
   grpc_slice small =
       grpc_slice_from_moved_string(grpc_core::UniquePtr<char>(small_ptr));
-  GPR_ASSERT(GRPC_SLICE_LENGTH(small) == strlen(kSmallStr));
-  GPR_ASSERT(GRPC_SLICE_START_PTR(small) !=
-             reinterpret_cast<uint8_t*>(small_ptr));
+  EXPECT_EQ(GRPC_SLICE_LENGTH(small), strlen(kSmallStr));
+  EXPECT_NE(GRPC_SLICE_START_PTR(small), reinterpret_cast<uint8_t*>(small_ptr));
   grpc_slice_unref_internal(small);
 
   // Large string should be move the reference.
   constexpr char kSLargeStr[] = "hello123456789123456789123456789";
   char* large_ptr = strdup(kSLargeStr);
   grpc_slice large =
       grpc_slice_from_moved_string(grpc_core::UniquePtr<char>(large_ptr));
-  GPR_ASSERT(GRPC_SLICE_LENGTH(large) == strlen(kSLargeStr));
-  GPR_ASSERT(GRPC_SLICE_START_PTR(large) ==
-             reinterpret_cast<uint8_t*>(large_ptr));
+  EXPECT_EQ(GRPC_SLICE_LENGTH(large), strlen(kSLargeStr));
+  EXPECT_EQ(GRPC_SLICE_START_PTR(large), reinterpret_cast<uint8_t*>(large_ptr));
   grpc_slice_unref_internal(large);
 
   // Moved buffer must respect the provided length not the actual length of the
   // string.
   large_ptr = strdup(kSLargeStr);
   small = grpc_slice_from_moved_buffer(grpc_core::UniquePtr<char>(large_ptr),
                                        strlen(kSmallStr));
-  GPR_ASSERT(GRPC_SLICE_LENGTH(small) == strlen(kSmallStr));
-  GPR_ASSERT(GRPC_SLICE_START_PTR(small) !=
-             reinterpret_cast<uint8_t*>(large_ptr));
+  EXPECT_EQ(GRPC_SLICE_LENGTH(small), strlen(kSmallStr));
+  EXPECT_NE(GRPC_SLICE_START_PTR(small), reinterpret_cast<uint8_t*>(large_ptr));
   grpc_slice_unref_internal(small);
 }
 
-void test_string_view_from_slice() {
+TEST(GrpcSliceTest, StringViewFromSlice) {
   constexpr char kStr[] = "foo";
   absl::string_view sv(
       grpc_core::StringViewFromSlice(grpc_slice_from_static_string(kStr)));
-  GPR_ASSERT(std::string(sv) == kStr);
+  EXPECT_EQ(std::string(sv), kStr);
```

Comment:
Not sure why the existing code was converting to `std::string` here.  I don't think that's necessary.

Reply:
Fixed